MonoGame.Framework.Content.Pipeline/Builder/PipelineManager.cs

```diff
@@ -716,65 +797,120 @@ public object ProcessContent(PipelineBuildEvent pipelineEvent)
                     }
-
-                    CleanContent(string.Empty, asset);
                 }
 
-                // Remove related output files (non-XNB files) that were copied to the output folder.
-                foreach (var asset in cachedEvent.BuildOutput)
+                if (cachedEvent != null)
                 {
-                    Logger.LogMessage("Cleaning {0}", asset);
-                    FileHelper.DeleteIfExists(asset);
-                }
-            }
+                    // Recursively clean additional (nested) assets.
+                    foreach (var asset in cachedEvent.BuildAsset)
+                    {
+                        string assetEventFilepath;
+                        var assetCachedEvent = LoadBuildEvent(asset, out assetEventFilepath);
 
-            Logger.LogMessage("Cleaning {0}", outputFilepath);
+                        if (assetCachedEvent == null)
+                        {
+                            Logger.LogMessage("Cleaning {0}", asset);
 
-            // Remove asset (.xnb file) from output folder.
-            FileHelper.DeleteIfExists(outputFilepath);
+                            // Remove asset (.xnb file) from output folder.
+                            FileHelper.DeleteIfExists(asset);
 
-            // Remove event file (.mgcontent file) from intermediate folder.
-            FileHelper.DeleteIfExists(eventFilepath);
+                            // Remove event file (.mgcontent file) from intermediate folder.
+                            FileHelper.DeleteIfExists(assetEventFilepath);
+                            continue;
+                        }
+
+                        CleanContent(asset);
+                    }
 
-            _pipelineBuildEvents.Remove(sourceFilepath);
+                    // Remove related output files (non-XNB files) that were copied to the output folder.
+                    foreach (var asset in cachedEvent.BuildOutput)
+                    {
+                        Logger.LogMessage("Cleaning {0}", asset);
+                        FileHelper.DeleteIfExists(asset);
+                    }
+
+                    // jcf: clean intermediate output???
+                }
+            }
+            finally
+            {
+                Logger.Unindent();
+                Logger.PopFile();
+            }
+
+            return clean;
         }
 
         private void WriteXnb(object content, PipelineBuildEvent pipelineEvent)
         {
             // Make sure the output directory exists.
             var outputFileDir = Path.GetDirectoryName(pipelineEvent.DestFile);
+            
+            var wrongCase = PathHelper.DirectoryExistsWithDifferentCase(outputFileDir);
+            if (wrongCase.HasValue && wrongCase.Value == true)
+            {
+                throw new Exception(string.Format("Failed to write xnb '{0}' because the path already exists in a different case!", pipelineEvent.DestFile));
```

Comment:
I am using this feature exactly for removing case present in the source asset name and folder structure from the output. If give the copy item "Data/foo.dat" the asset name "data/foo.dat" and it says it built successfully, but when I look in bin/ its actually "Data/foo.dat" then it hasn't done what I asked and silently failed in a way that I wont realize until I crash trying to File.Open("data/foo.dat") on an actual ps4 (actually its even worse, it wont fail until I try it on an installed package).